synapse/handlers/message.py

```diff
@@ -888,9 +888,20 @@ def persist_and_notify_client_event(
         yield self.base_handler.maybe_kick_guest_users(event, context)
 
         if event.type == EventTypes.CanonicalAlias:
-            # Check the alias is acually valid (at this time at least)
+            # Validate a newly added alias or newly added alt_aliases.
+
+            original_event = yield self.state.get_current_state(
+                event.room_id, EventTypes.CanonicalAlias, ""
+            )
```

Comment:
the problem here is that the room might have moved on since the event was created: we're not interested in the current state, so much as the state at `context`.

It looks like `compute_event_context` populates `event.unsigned.replaces_state` with the id of the event being replaced, if it exists. I think you should be able to use that to find the event of interest.

Reply:
I believe I changed this in 2bfd4d8034fef049b40bcaffd5733ebb330ef475, but I'm now 100% sure on the API I chose instead (`resolve_state_groups_for_events`). Is there an easier way to get a previous event?

Reply:
sure, there's a `store.get_event` method.

Reply:
Thanks! I switched that out.